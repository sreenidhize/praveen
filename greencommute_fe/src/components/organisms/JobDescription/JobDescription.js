import React from "react";
import { Box, Grid, makeStyles } from "@material-ui/core";
import JobDetails from "../../molecules/jobDetails/JobDetails";
import JobDetailsHeader from "../../molecules/JobDetailsHeader/JobDetialsHeader";
import PropTypes from "prop-types";
import Button from "../../atom/button/Button";

const useStyle = makeStyles({
  root: {
    position: "relative",
  },
  boxClass: {
    position: "absolute",
    left: "0",
    bottom: "3.2%",
    backgroundImage:
      "linear-gradient(to bottom, rgba(255, 255, 255, 0), #ffffff, #ffffff)",
  },
  buttonClass: {
    position: "absolute",
    top: "55%",
    left: "15%",
    borderRadius: "10px",
    width: "208px",
    height: "50px",
    padding: " 16px 15.6px 16px 14.4px",
    marginRight: "10px",
  },
});
function JobDescription(props) {
  const classes = useStyle();

  const {
    headerProps,
    description,
    skills,
    handleOnSave,
    button1Label,
    onClick2,
    onClick3,
  } = props;
  return (
    <Box className={classes.root}>
      <Grid container direction="column" spacing={6}>
        <Grid item>
          <JobDetailsHeader
            {...{ headerProps, handleOnSave, onClick2, button1Label }}
          />
        </Grid>
        <Grid item>
          <Box marginLeft="0.2%">
            <JobDetails description={description} skills={skills} />
          </Box>
        </Grid>
        <Box height="17.5%" width="100%" className={classes.boxClass}>
          <Button
            label="Green Commute Routes"
            color="primary"
            type="long"
            variant="contained"
            onClick={onClick3}
            className={classes.buttonClass}
            classes
          />
        </Box>
      </Grid>
    </Box>
  );
}

JobDescription.propTypes = {
  headerProps: PropTypes.object,
  description: PropTypes.string,
  skills: PropTypes.array,
};

JobDescription.defaultProps = {};

export default JobDescription;
