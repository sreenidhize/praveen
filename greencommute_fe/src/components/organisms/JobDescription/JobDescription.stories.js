import { Grid } from "@material-ui/core";
import React from "react";
import JobDescription from "./JobDescription";

const JobDetailsHeaderStory = {
  title: "Organisms/DescriptionDetails",
  component: JobDescription,
  decorators: [
    (Story) => (
      <Grid style={{ margin: "1px" }}>
        <Story />
      </Grid>
    ),
  ],
};

const Template = (args) => <JobDescription {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  headerProps: {
    src: "https://upload.wikimedia.org/wikipedia/commons/4/44/BMW.svg",
    alt: "BMW company",
    jobTitle: "User Experience Designer",
    companyName: "BMW",
    location: "Hi-Tech City, Telangana",
  },

  description:
    "Open Text is seeking a talented, personable interaction designer who can assist the User Experience Design team by working with other designers and development teams on a variety of projects. The OpenText User Experience Design group is a distributed multi-disciplinary team of professionals that are responsible for enhancing the UX of the companyâ€™s collective product suites worldwide.",
  skills: [
    "High level of proficiency with leading UX Design software packages, such as Axure, Sketch, InVision, or Experience Design including the core Adobe Creative Suite products.",
    "Excellent written and oral communication and presentation skills",
    "Excellent written and oral communication and presentation skills",
    "Excellent written and oral communication",
  ],
};

export default JobDetailsHeaderStory;
