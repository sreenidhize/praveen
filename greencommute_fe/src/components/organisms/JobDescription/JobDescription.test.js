import { render, screen } from "@testing-library/react";
import React from "react";
import JobDescription from "./JobDescription";
import userEvent from "@testing-library/user-event";

describe("JobDescription renders correctly", () => {
  const headerProps = {
    src: "https://upload.wikimedia.org/wikipedia/commons/4/44/BMW.svg",
    alt: "BMW company",
    jobTitle: "User Experience Designer",
    companyName: "BMW",
    location: "Hi-Tech City, Telangana",
  };
  const description =
    "Open Text is seeking a talented, personable interaction designer who can assist the User Experience Design team by working with other designers and development teams on a variety of projects. The OpenText User Experience Design group is a distributed multi-disciplinary team of professionals that are responsible for enhancing the UX of the companyâ€™s collective product suites worldwide.";

  const skills = [
    "High level of proficiency with leading UX Design software packages, such as Axure, Sketch, InVision, or Experience Design including the core Adobe Creative Suite products.",
    "Excellent written and oral communication and presentation skills",
    "Excellent written and oral communication and presentation skills",
  ];
  test("should match snapshot", () => {
    const { asFragment } = render(
      <JobDescription
        headerProps={headerProps}
        description={description}
        skills={skills}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("JobDescription should render logo and buttons should have onClick", () => {
    const onClick1 = jest.fn();
    const onClick2 = jest.fn();
    const onClick3 = jest.fn();
    const { container } = render(
      <JobDescription
        headerProps={headerProps}
        description={description}
        skills={skills}
        handleOnSave={onClick1}
        onClick2={onClick2}
        onClick3={onClick3}
      />
    );
    expect(container.getElementsByTagName("img").length).toBe(1);
    expect(container.getElementsByTagName("h4").length).toBe(3);
    expect(container.getElementsByTagName("button").length).toBe(3);
    userEvent.click(screen.getByText("Save"));
    expect(onClick1).toHaveBeenCalled();
    userEvent.click(screen.getByText("Apply"));
    expect(onClick2).toHaveBeenCalled();
    userEvent.click(screen.getByText("Green Commute Routes"));
    expect(onClick2).toHaveBeenCalled();
  });
});
