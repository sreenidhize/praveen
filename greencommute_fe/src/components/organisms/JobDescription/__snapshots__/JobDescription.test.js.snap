// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobDescription renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-4 makeStyles-root-1"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6 MuiGrid-direction-xs-column"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-direction-xs-column"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiGrid-root makeStyles-root-7"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circle makeStyles-smallMedia-8"
                >
                  <img
                    alt="BMW company"
                    class="MuiAvatar-img"
                    src="https://upload.wikimedia.org/wikipedia/commons/4/44/BMW.svg"
                  />
                </div>
              </div>
              <div
                class="MuiGrid-root makeStyles-columnMargin-6 MuiGrid-container MuiGrid-direction-xs-column"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <h4
                    class="MuiTypography-root makeStyles-primary-9 MuiTypography-h4"
                  >
                    User Experience Designer
                  </h4>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <h4
                    class="MuiTypography-root makeStyles-secondary-10 MuiTypography-h4"
                  >
                    BMW
                  </h4>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <h4
                    class="MuiTypography-root makeStyles-secondary-10 MuiTypography-h4"
                  >
                    Hi-Tech City, Telangana
                  </h4>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
            >
              <div
                style="display: flex;"
              >
                <div
                  class="MuiBox-root MuiBox-root-15"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-base-16 makeStyles-small-17 MuiButton-outlinedPrimary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Save
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-22"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-base-16 makeStyles-small-17 MuiButton-containedPrimary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Apply
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiBox-root MuiBox-root-23"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
            style="width: 390px;"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <hr
                class="MuiDivider-root makeStyles-hrBottom-29"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root makeStyles-headingClass-24 MuiTypography-body1"
              >
                Description
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <h6
                class="MuiTypography-root makeStyles-descClass-27 MuiTypography-h6"
              >
                Open Text is seeking a talented, personable interaction designer who can assist the User Experience Design team by working with other designers and development teams on a variety of projects. The OpenText User Experience Design group is a distributed multi-disciplinary team of professionals that are responsible for enhancing the UX of the companyâ€™s collective product suites worldwide.
              </h6>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <hr
                class="MuiDivider-root makeStyles-hrTop-28"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                class="MuiTypography-root makeStyles-headingClass-24 MuiTypography-body1"
              >
                What it takes
              </p>
              <ul
                class="makeStyles-ulClass-25"
              >
                <li
                  class="makeStyles-liClass-26"
                >
                  <h6
                    class="MuiTypography-root makeStyles-descClass-27 MuiTypography-h6"
                  >
                    High level of proficiency with leading UX Design software packages, such as Axure, Sketch, InVision, or Experience Design including the core Adobe Creative Suite products.
                  </h6>
                </li>
                <li
                  class="makeStyles-liClass-26"
                >
                  <h6
                    class="MuiTypography-root makeStyles-descClass-27 MuiTypography-h6"
                  >
                    Excellent written and oral communication and presentation skills
                  </h6>
                </li>
                <li
                  class="makeStyles-liClass-26"
                >
                  <h6
                    class="MuiTypography-root makeStyles-descClass-27 MuiTypography-h6"
                  >
                    Excellent written and oral communication and presentation skills
                  </h6>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-30 makeStyles-boxClass-2"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonClass-3 MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Green Commute Routes
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
